sklearn/compose/tests/test_target.py

```diff
@@ -393,3 +393,17 @@ def test_transform_target_regressor_pass_extra_predict_parameters():
     regr.fit(X, y)
     regr.predict(X, check_input=False)
     assert regr.regressor_.predict_called
+
+
+@pytest.mark.filterwarnings("error")
```

Comment:
Did you find this pattern used in other places in our test? Personally I would maybe find it a bit less clear than the [recommendation from Pytest](https://docs.pytest.org/en/7.1.x/how-to/capture-warnings.html#additional-use-cases-of-warnings-in-tests) for checking that no warnings is emitted:
```py
with warnings.catch_warnings():
    warnings.simplefilter("error")
    ...
```

If I had to give a small justification for it, I would say that the code checking that there is no warning is closer to code potentially creating the warning which maybe helps making the intent a bit clearer ...

Reply:
Yes, I see what you mean. `@pytest.mark.filterwarnings("error")` is actually not used anywhere else in scikit-learn. I was finding it by googling how to make the test not pass, when a warning is raised.

Let's better use `warnings.catch_warnings` instead.